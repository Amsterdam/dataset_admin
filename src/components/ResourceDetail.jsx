import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import Form from 'react-jsonschema-form';
import extraFields from 'react-jsonschema-form-extras';

import { fetchDataset } from '../actions/dataset';
import localFields from '../fields';
import widgets from '../widgets';

import '../../node_modules/react-day-picker/lib/style.css';
import './dcatd-form.scss';

const fields = {
  ...extraFields,
  ...localFields
};

const mapStateToProps = state => ({
  dataset: state.dataset
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchDataset
}, dispatch);

class ResourceDetail extends Component {
  constructor(props) {
    super(props);

    this.state = {
      uiResource: props.uiResource,
      formData: props.formData
    };

    this.setVisibilityOfFields = this.setVisibilityOfFields.bind(this);
  }

  componentDidMount() {
    if (this.hasDataset()) {
      // this.props.fetchDataset(this.props.id);
    }
  }

  setVisibilityOfFields(event) {
    if (event.formData['ams:distributionType'] === 'file') {
      this.showField('dct:format', 'select');
    } else {
      this.hideField('dct:format');
    }

    if (event.formData['ams:distributionType'] === 'api') {
      this.showField('ams:serviceType', 'select');
    } else {
      this.hideField('ams:serviceType');
    }

    this.setState({
      formData: {
        ...event.formData
      }
    });
  }

  showField(name, type = 'text') {
    this.setState({
      uiResource: {
        ...this.state.uiResource,
        [name]: {
          'ui:widget': type
        }
      }
    });
  }

  hideField(name) {
    this.setState({
      uiResource: {
        ...this.state.uiResource,
        [name]: {
          'ui:widget': 'hidden'
        }
      }
    });
  }

  hasDataset() {
    return this.props.id && this.props.id !== 'new';
  }

  render() {
    const { formData } = this.state;
    return (
      <div>
        <Form
          className="dcatd-form resource-form"
          schema={this.props.schema}
          formData={formData}
          widgets={widgets}
          fields={fields}
          uiSchema={this.state.uiResource}
          noHtml5Validate
          showErrorList={false}
          onChange={event => this.setVisibilityOfFields(event)}
        />
      </div>
    );
  }
}

ResourceDetail.defaultProps = {
  formData: {},
  id: null
};

ResourceDetail.propTypes = {
  formData: PropTypes.object,
  schema: PropTypes.object.isRequired,
  // uiDataset: PropTypes.object.isRequired,
  uiResource: PropTypes.object.isRequired,
  id: PropTypes.string
  // dataset: PropTypes.object.isRequired

  // fetchDataset: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(ResourceDetail);
