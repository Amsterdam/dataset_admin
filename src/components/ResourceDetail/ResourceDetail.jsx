import React, { Component } from 'react';
import PropTypes from 'prop-types';
import defer from 'lodash.defer';

import Form from 'react-jsonschema-form';
import extraFields from 'react-jsonschema-form-extras';

import Modal from '../Modal/Modal';
import transformErrors from '../../services/transform-errors/transform-errors';
import scrollToError from '../../services/scroll-to-error/scroll-to-error';
import localFields from '../../fields';
import widgets from '../../widgets';

import '../../../node_modules/react-day-picker/lib/style.css';
import '../dcatd-form.scss';

const fields = {
  ...extraFields,
  ...localFields
};

class ResourceDetail extends Component {
  constructor(props) {
    super(props);

    this.state = {
      uiResource: props.uiResource,
      formData: props.formData
    };

    this.setVisibilityOfFields = this.setVisibilityOfFields.bind(this);
    this.setFieldState = this.setFieldState.bind(this);
    this.setResourceSpecs = this.setResourceSpecs.bind(this);
    this.hasResource = this.hasResource.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleCancel = this.handleCancel.bind(this);
  }

  componentWillReceiveProps(props) {
    this.setState({
      uiResource: { ...props.uiResource },
      formData: { ...props.formData }
    });

    defer(() => {
      this.setVisibilityOfFields(this.state.formData);
    });
  }

  setVisibilityOfFields(formData) {
    if (formData['ams:distributionType'] === 'file') {
      this.setFieldState('dct:format', 'select');
    } else {
      this.setFieldState('dct:format', 'hidden');
    }

    if (formData['ams:distributionType'] === 'api') {
      this.setFieldState('ams:serviceType', 'select');
    } else {
      this.setFieldState('ams:serviceType', 'hidden');
    }

    this.setState({
      formData: {
        ...formData
      }
    });
  }

  setFieldState(name, value) {
    this.setState({
      uiResource: {
        ...this.state.uiResource,
        [name]: {
          ...this.state.uiResource[name],
          'ui:widget': value
        }
      }
    });
  }

  setResourceSpecs(values) {
    this.setState({
      formData: {
        ...this.state.formData,
        ...values
      }

    });
  }

  hasResource() {
    return this.props.formData['@id'];
  }

  handleSubmit(formData) {
    this.props.handleResourceToDataset(formData);
    this.props.onEmptyResource();
  }

  handleCancel() {
    this.props.onEmptyResource();
  }

  render() {
    const { formData } = this.state;
    return (
      <div>
        <Form
          className="dcatd-form resource-form"
          idPrefix="resource"
          schema={this.props.schema}
          formData={formData}
          formContext={{
            setResourceSpecs: this.setResourceSpecs
          }}
          widgets={widgets}
          fields={fields}
          uiSchema={this.state.uiResource}
          noHtml5Validate
          showErrorList={false}
          transformErrors={transformErrors}
          onError={scrollToError}
          onSubmit={event => this.handleSubmit(event.formData)}
          onChange={event => this.setVisibilityOfFields(event.formData)}
        >
          <div>
            <button
              className="dcatd-form-button dcatd-form-button-submit"
              type="submit"
            >
              Opslaan</button>
            <button
              onClick={() => this.props.onEmptyResource()}
              className="dcatd-form-button dcatd-form-button-cancel"
              type="button"
            >
              Annuleren</button>
            {this.hasResource() ?
              <Modal
                ref={(ref) => { this.modal = ref; }}
                content="Door de resource te verwijderen, gaan alle gegevens verloren."
                actionLabel="Resource verwijderen"
                trigger={(
                  <button
                    onClick={() => this.modal.handleShowState(true)}
                    type="button"
                    className="dcatd-form-button dcatd-form-button-remove"
                  >
                    Resource verwijderen
                  </button>
                )}
                onProceed={() => {
                  this.handleSubmit({
                    '@id': formData['@id'],
                    'dcat:accessURL': 'remove'
                  });
                }}
              />
              : ''}
          </div>
        </Form>
      </div>
    );
  }
}

ResourceDetail.defaultProps = {
  formData: {}
};

ResourceDetail.propTypes = {
  formData: PropTypes.object,
  schema: PropTypes.object.isRequired,
  uiResource: PropTypes.object.isRequired,

  handleResourceToDataset: PropTypes.func.isRequired,
  onEmptyResource: PropTypes.func.isRequired
};

export default ResourceDetail;
