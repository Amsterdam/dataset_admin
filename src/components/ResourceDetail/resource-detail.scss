@import "../../styling/config.scss";

.resource-form {
  &__back {
    @extend %heavy-font;
    background-color: $white;
    border-bottom: 1px solid $pinkish-grey;
    border-left-style: none;
    border-right-style: none;
    border-top-style: none;
    margin-top: -14px;
    padding: 9px 0 8px 30px;
    position: relative;
    text-align: left;
    width: 100%;

    &::before {
      @extend %icon-arrow-left;
      background-size: 100%;
      content: "";
      display: inline-block;
      height: 22px;
      left: 0;
      position: absolute;
      width: 22px;
    }
  }

  &__disclaimer {
    color: $brownish-grey;
    font-size: 14px;
    margin-top: 6px;
    width: 50%;
  }
}
