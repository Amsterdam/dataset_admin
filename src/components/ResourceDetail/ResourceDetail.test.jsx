import React from 'react';

import ResourceDetail from './ResourceDetail';

describe('The ResourceDetail component', () => {
  let wrap;
  let datasetTitle;
  let onSetResourceToDatasetSpy;
  let onEmptyResourceSpy;
  let setModalSpy;

  const mockSchema = {
    type: 'array',
    default: [],
    items: {
      properties: {
        'ams:resourceType': {
          enum: ['data', 'doc', 'vis', 'app'],
          enumNames: ['Data', 'Documentatie', 'Visualisatie', 'Voorbeeldtoepassing']
        }
      }
    }
  };

  beforeEach(() => {
    setModalSpy = jest.fn();
    onEmptyResourceSpy = jest.fn();
    onSetResourceToDatasetSpy = jest.fn();
    datasetTitle = '';
    wrap = shallow((
      <ResourceDetail
        schema={mockSchema}
        uiResource={{}}
        formData={{}}
        datasetTitle={datasetTitle}
        onSetResourceToDataset={onSetResourceToDatasetSpy}
        onEmptyResource={onEmptyResourceSpy}
        setModal={setModalSpy}
      />
    ));
  });

  describe('renders component', () => {
    it('renders default props', () => {
      expect(wrap).toMatchSnapshot();
    });
  });

  // describe('triggering events', () => {
  //   let setResourceDetailSpy;
  //   let onProceedSpy;
  //
  //   beforeEach(() => {
  //     setResourceDetailSpy = jest.fn();
  //     onProceedSpy = jest.fn();
  //   });
  //
  //   it('clicking submit closes modal and triggers onProceed', () => {
  //     wrap.setProps({
  //       modal: {
  //         onProceed: onProceedSpy
  //       },
  //       setResourceDetail: setResourceDetailSpy
  //     });
  //
  //     wrap.find('.dcatd-form-button-submit').simulate('click');
  //
  //     expect(onProceedSpy).toHaveBeenCalled();
  //     expect(setResourceDetailSpy).toHaveBeenCalledWith({ open: false });
  //   });
  //
  //   it('clicking cancel closes modal', () => {
  //     wrap.setProps({
  //       setResourceDetail: setResourceDetailSpy
  //     });
  //
  //     wrap.find('.dcatd-form-button-cancel').simulate('click');
  //
  //     expect(onProceedSpy).not.toHaveBeenCalled();
  //     expect(setResourceDetailSpy).toHaveBeenCalledWith({ open: false });
  //   });
  // });
});
