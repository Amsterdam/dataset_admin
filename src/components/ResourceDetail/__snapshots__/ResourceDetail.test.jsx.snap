// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ResourceDetail component renders component renders default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResourceDetail
    datasetTitle=""
    formData={Object {}}
    onEmptyResource={[MockFunction]}
    onSetResourceToDataset={[MockFunction]}
    schema={
      Object {
        "default": Array [],
        "items": Object {
          "properties": Object {
            "ams:resourceType": Object {
              "enum": Array [
                "data",
                "doc",
                "vis",
                "app",
              ],
              "enumNames": Array [
                "Data",
                "Documentatie",
                "Visualisatie",
                "Voorbeeldtoepassing",
              ],
            },
          },
        },
        "type": "array",
      }
    }
    setModal={[MockFunction]}
    uiResource={Object {}}
    uploadStatus="idle"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="resource-form__back"
          onClick={[Function]}
        >
          
        </button>,
        <h1
          className="resource-title"
        >
          Resource 
          toevoegen
        </h1>,
        <Form
          ErrorList={[Function]}
          className="dcatd-form resource-form"
          fields={
            Object {
              "altInput": [Function],
              "asyncTypeahead": [Function],
              "collapsible": [Function],
              "compositeArray": [Function],
              "dropdown": [Function],
              "rdp": [Function],
              "resources": [Function],
              "rte": [Function],
              "table": [Function],
              "typeahead": [Function],
            }
          }
          formContext={
            Object {
              "setResourceSpecs": [Function],
              "setUploadStatus": [Function],
            }
          }
          formData={Object {}}
          idPrefix="resource"
          liveValidate={false}
          noHtml5Validate={true}
          noValidate={false}
          onChange={[Function]}
          onError={[Function]}
          onSubmit={[Function]}
          safeRenderCompletion={false}
          schema={
            Object {
              "default": Array [],
              "items": Object {
                "properties": Object {
                  "ams:resourceType": Object {
                    "enum": Array [
                      "data",
                      "doc",
                      "vis",
                      "app",
                    ],
                    "enumNames": Array [
                      "Data",
                      "Documentatie",
                      "Visualisatie",
                      "Voorbeeldtoepassing",
                    ],
                  },
                },
              },
              "type": "array",
            }
          }
          showErrorList={false}
          transformErrors={[Function]}
          uiSchema={Object {}}
          widgets={
            Object {
              "file": [Function],
              "markdown": [Function],
            }
          }
        >
          <div>
            <button
              className="dcatd-form-button dcatd-form-button-submit"
              type="submit"
            >
              OK en terug
            </button>
            <button
              className="dcatd-form-button dcatd-form-button-cancel"
              onClick={[Function]}
              type="button"
            >
              Annuleren
            </button>
            
            <div
              className="resource-form__disclaimer"
            >
              Let op: na het klikken op OK is de resource nog niet opgeslagen. Pas tijdens het opslaan van de dataset wordt ook de resource opgeslagen.
            </div>
          </div>
        </Form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resource-form__back",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Resource ",
            "toevoegen",
          ],
          "className": "resource-title",
        },
        "ref": null,
        "rendered": Array [
          "Resource ",
          "toevoegen",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ErrorList": [Function],
          "children": <div>
            <button
              className="dcatd-form-button dcatd-form-button-submit"
              type="submit"
            >
              OK en terug
            </button>
            <button
              className="dcatd-form-button dcatd-form-button-cancel"
              onClick={[Function]}
              type="button"
            >
              Annuleren
            </button>
            
            <div
              className="resource-form__disclaimer"
            >
              Let op: na het klikken op OK is de resource nog niet opgeslagen. Pas tijdens het opslaan van de dataset wordt ook de resource opgeslagen.
            </div>
          </div>,
          "className": "dcatd-form resource-form",
          "fields": Object {
            "altInput": [Function],
            "asyncTypeahead": [Function],
            "collapsible": [Function],
            "compositeArray": [Function],
            "dropdown": [Function],
            "rdp": [Function],
            "resources": [Function],
            "rte": [Function],
            "table": [Function],
            "typeahead": [Function],
          },
          "formContext": Object {
            "setResourceSpecs": [Function],
            "setUploadStatus": [Function],
          },
          "formData": Object {},
          "idPrefix": "resource",
          "liveValidate": false,
          "noHtml5Validate": true,
          "noValidate": false,
          "onChange": [Function],
          "onError": [Function],
          "onSubmit": [Function],
          "safeRenderCompletion": false,
          "schema": Object {
            "default": Array [],
            "items": Object {
              "properties": Object {
                "ams:resourceType": Object {
                  "enum": Array [
                    "data",
                    "doc",
                    "vis",
                    "app",
                  ],
                  "enumNames": Array [
                    "Data",
                    "Documentatie",
                    "Visualisatie",
                    "Voorbeeldtoepassing",
                  ],
                },
              },
            },
            "type": "array",
          },
          "showErrorList": false,
          "transformErrors": [Function],
          "uiSchema": Object {},
          "widgets": Object {
            "file": [Function],
            "markdown": [Function],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="dcatd-form-button dcatd-form-button-submit"
                type="submit"
              >
                OK en terug
              </button>,
              <button
                className="dcatd-form-button dcatd-form-button-cancel"
                onClick={[Function]}
                type="button"
              >
                Annuleren
              </button>,
              "",
              <div
                className="resource-form__disclaimer"
              >
                Let op: na het klikken op OK is de resource nog niet opgeslagen. Pas tijdens het opslaan van de dataset wordt ook de resource opgeslagen.
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "OK en terug",
                "className": "dcatd-form-button dcatd-form-button-submit",
                "type": "submit",
              },
              "ref": null,
              "rendered": "OK en terug",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Annuleren",
                "className": "dcatd-form-button dcatd-form-button-cancel",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Annuleren",
              "type": "button",
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Let op: na het klikken op OK is de resource nog niet opgeslagen. Pas tijdens het opslaan van de dataset wordt ook de resource opgeslagen.",
                "className": "resource-form__disclaimer",
              },
              "ref": null,
              "rendered": "Let op: na het klikken op OK is de resource nog niet opgeslagen. Pas tijdens het opslaan van de dataset wordt ook de resource opgeslagen.",
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="resource-form__back"
            onClick={[Function]}
          >
            
          </button>,
          <h1
            className="resource-title"
          >
            Resource 
            toevoegen
          </h1>,
          <Form
            ErrorList={[Function]}
            className="dcatd-form resource-form"
            fields={
              Object {
                "altInput": [Function],
                "asyncTypeahead": [Function],
                "collapsible": [Function],
                "compositeArray": [Function],
                "dropdown": [Function],
                "rdp": [Function],
                "resources": [Function],
                "rte": [Function],
                "table": [Function],
                "typeahead": [Function],
              }
            }
            formContext={
              Object {
                "setResourceSpecs": [Function],
                "setUploadStatus": [Function],
              }
            }
            formData={Object {}}
            idPrefix="resource"
            liveValidate={false}
            noHtml5Validate={true}
            noValidate={false}
            onChange={[Function]}
            onError={[Function]}
            onSubmit={[Function]}
            safeRenderCompletion={false}
            schema={
              Object {
                "default": Array [],
                "items": Object {
                  "properties": Object {
                    "ams:resourceType": Object {
                      "enum": Array [
                        "data",
                        "doc",
                        "vis",
                        "app",
                      ],
                      "enumNames": Array [
                        "Data",
                        "Documentatie",
                        "Visualisatie",
                        "Voorbeeldtoepassing",
                      ],
                    },
                  },
                },
                "type": "array",
              }
            }
            showErrorList={false}
            transformErrors={[Function]}
            uiSchema={Object {}}
            widgets={
              Object {
                "file": [Function],
                "markdown": [Function],
              }
            }
          >
            <div>
              <button
                className="dcatd-form-button dcatd-form-button-submit"
                type="submit"
              >
                OK en terug
              </button>
              <button
                className="dcatd-form-button dcatd-form-button-cancel"
                onClick={[Function]}
                type="button"
              >
                Annuleren
              </button>
              
              <div
                className="resource-form__disclaimer"
              >
                Let op: na het klikken op OK is de resource nog niet opgeslagen. Pas tijdens het opslaan van de dataset wordt ook de resource opgeslagen.
              </div>
            </div>
          </Form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resource-form__back",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Resource ",
              "toevoegen",
            ],
            "className": "resource-title",
          },
          "ref": null,
          "rendered": Array [
            "Resource ",
            "toevoegen",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ErrorList": [Function],
            "children": <div>
              <button
                className="dcatd-form-button dcatd-form-button-submit"
                type="submit"
              >
                OK en terug
              </button>
              <button
                className="dcatd-form-button dcatd-form-button-cancel"
                onClick={[Function]}
                type="button"
              >
                Annuleren
              </button>
              
              <div
                className="resource-form__disclaimer"
              >
                Let op: na het klikken op OK is de resource nog niet opgeslagen. Pas tijdens het opslaan van de dataset wordt ook de resource opgeslagen.
              </div>
            </div>,
            "className": "dcatd-form resource-form",
            "fields": Object {
              "altInput": [Function],
              "asyncTypeahead": [Function],
              "collapsible": [Function],
              "compositeArray": [Function],
              "dropdown": [Function],
              "rdp": [Function],
              "resources": [Function],
              "rte": [Function],
              "table": [Function],
              "typeahead": [Function],
            },
            "formContext": Object {
              "setResourceSpecs": [Function],
              "setUploadStatus": [Function],
            },
            "formData": Object {},
            "idPrefix": "resource",
            "liveValidate": false,
            "noHtml5Validate": true,
            "noValidate": false,
            "onChange": [Function],
            "onError": [Function],
            "onSubmit": [Function],
            "safeRenderCompletion": false,
            "schema": Object {
              "default": Array [],
              "items": Object {
                "properties": Object {
                  "ams:resourceType": Object {
                    "enum": Array [
                      "data",
                      "doc",
                      "vis",
                      "app",
                    ],
                    "enumNames": Array [
                      "Data",
                      "Documentatie",
                      "Visualisatie",
                      "Voorbeeldtoepassing",
                    ],
                  },
                },
              },
              "type": "array",
            },
            "showErrorList": false,
            "transformErrors": [Function],
            "uiSchema": Object {},
            "widgets": Object {
              "file": [Function],
              "markdown": [Function],
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="dcatd-form-button dcatd-form-button-submit"
                  type="submit"
                >
                  OK en terug
                </button>,
                <button
                  className="dcatd-form-button dcatd-form-button-cancel"
                  onClick={[Function]}
                  type="button"
                >
                  Annuleren
                </button>,
                "",
                <div
                  className="resource-form__disclaimer"
                >
                  Let op: na het klikken op OK is de resource nog niet opgeslagen. Pas tijdens het opslaan van de dataset wordt ook de resource opgeslagen.
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "OK en terug",
                  "className": "dcatd-form-button dcatd-form-button-submit",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "OK en terug",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Annuleren",
                  "className": "dcatd-form-button dcatd-form-button-cancel",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Annuleren",
                "type": "button",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Let op: na het klikken op OK is de resource nog niet opgeslagen. Pas tijdens het opslaan van de dataset wordt ook de resource opgeslagen.",
                  "className": "resource-form__disclaimer",
                },
                "ref": null,
                "rendered": "Let op: na het klikken op OK is de resource nog niet opgeslagen. Pas tijdens het opslaan van de dataset wordt ook de resource opgeslagen.",
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
